src/main/java/stsjorbsmod/patches/FilterRandomCardGenerationPatch.java

```diff
@@ -0,0 +1,75 @@
+package stsjorbsmod.patches;
+
+import com.evacipated.cardcrawl.modthespire.lib.*;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.random.Random;
+import javassist.CtBehavior;
+
+import java.util.ArrayList;
+
+import static stsjorbsmod.characters.Wanderer.Enums.PERSISTENT_POSITIVE_EFFECT;
+
+public class FilterRandomCardGenerationPatch {
+    private static void RemovePersistentPositiveEffects(ArrayList list) {
+        list.removeIf(o -> ((AbstractCard) o).hasTag(PERSISTENT_POSITIVE_EFFECT));
+    }
+
+    @SpirePatch(
+            clz = AbstractDungeon.class,
+            method = "returnTrulyRandomCardInCombat",
+            paramtypez = { }
+    )
+    public static class AbstractDungeon_returnTrulyRandomCardInCombat_1 {
+        @SpireInsertPatch(
+                locator = Locator.class,
+                localvars = "list"
+        )
+        @SuppressWarnings("unchecked")
+        public static void patch(@ByRef ArrayList[] list) {
```

Comment:
This doesn't need to be byref; you're modifying the contents of the list, not pointing it to a new list.

Reply:
(same elsewhere)

Reply:
It's hilarious/sad that I knew that for the helper function.

Reply:
I just remembered what the kids say: smh.
I don't have an emoji lexicon to remember anything from.